app/Http/Controllers/Auth/LoginController.php

```diff
@@ -74,6 +74,13 @@ public function showLoginForm(Request $request)
             return redirect()->intended('/');
         }
 
+        //If the environment is set to ALWAYS require SAML, go straight to the SAML route.
+        //We don't need to check other settings, as this should override those.
+        if(config('REQUIRE_SAML'))
```

Comment:
Almost there :)

If we're using the `config('foo.bar')` convention, we add the `env()` to the config file (which you did in `config/app.php`), you'd then refer to it by the config filename (minus the `.php` part) and then the new variable you created in the config, in this case `if (config('app.require_saml'))`.

(Also please always add a space after `if` and `else`s. It's a small thing, but I'll only fix it later if you don't. :)

Reply:
Clearly, my Laravel is a bit rusty. ;)

Reply:
No, no, not at all! This is a really subtle area of difference that's super-duper easy to get turned around. And @snipe has her own conventions that she likes for the codebase - which I have absolutely been guilty of breaking, and I work here :)

Reply:
All fixed. I will let you resolve it this time in case there is still an error. ;)